plugins/modules/mount.py

```diff
@@ -813,6 +827,9 @@ def main():
                             if not (ex.errno == errno.EEXIST and os.path.isdir(b_curpath)):
                                 raise
 
+                if mode is not None:
+                    os.chmod(name, int(mode))
```

Comment:
I have couple of comments over here -

- Move this block out try-except to its own try-except. So that we don't mask the original exception.
- Please use [set_mode_if_different](https://github.com/ansible/ansible/blob/cf3a304ce1711ed9a19549fbf58c004acd47c06a/lib/ansible/module_utils/basic.py#L885) instead of `chmod`.
- Other modules provides `mode` such as `u+rw,g-wx,o-rwx` or `u=rw,g=r,o=r`, please add this support as well.

Reply:
Thank you for your advise.
I fixed my code based on the comments.